src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -454,9 +454,27 @@ export class RemoveAction extends AbstractSearchAndReplaceAction {
 	}
 
 	override run(): Promise<any> {
-		const currentFocusElement = this.viewer.getFocus()[0];
-		const nextFocusElement = !currentFocusElement || currentFocusElement instanceof SearchResult || elementIsEqualOrParent(currentFocusElement, this.element) ?
-			this.getElementToFocusAfterRemoved(this.viewer, this.element) :
+
+		let elementsToRemove: RenderableMatch[] = [];
+		const currentSelection: RenderableMatch[] = [];
+
+		this.viewer.getSelection().forEach((elem) => {
+			currentSelection.push(<RenderableMatch>elem);
```

Comment:
What's the purpose of this, isn't it just copying from one array to another array?

Reply:
It's casting the array contents to `RenderableMatch` elements and pushing them to a new array

Reply:
The cast doesn't do anything at runtime, so a new array isn't needed, you can just cast. But I'm curious why the cast is necessary

Reply:
I believe it had to do with a function needing the matches as `RenderableMatch`s as arguments. Here, I'm casting them before I needed them as arguments

Reply:
Sure, but are they always RenderableMatches? If the type signature of getSelection says that it could return something else, then it may not be a safe cast

Reply:
good point, I'll take a look